### ISSUE TITLE
Add support for testing specific mutants

### ISSUE THREAD

**Issue by Aimless321** [stryker-mutator/editor-plugins#10]
Currently, it is possible to specify which blocks of code need to be tested using mutation ranges, but there's no way to specify which individual mutants within those blocks should be tested.

## Use case
In the Stryker Cloud Orchestrator, a mutation test run is split into tasks, where each task contains a set of mutants to be tested. These tasks are optimized based on test coverage to allow multiple mutants to be tested in the same task as long as there is no overlap in their test coverage.

To support this use case, it would be helpful to have a mechanism to specify exactly which mutants should be tested, rather than testing all mutants within a given code block.

Would it be possible to do this based on the `mutatorName` in combination with the current ranges?

**Comment by rouke-broersma** [comment]
In our use case mutant IDs are stable, so we could also simply optionally allow mutant ID in MSP.

**Comment by Aimless321** [comment]
To add another example where the current implementation could cause issues.

These mutants are returned by a discover:
```json
{
  "id": "1",
  "location": {
    "start": {
      "line": 7,
      "column": 41
    },
    "end": {
      "line": 9,
      "column": 4
    }
  },
  "description": null,
  "mutatorName": "BlockStatement",
  "replacement": "{}"
},
{
  "id": "2",
  "location": {
    "start": {
      "line": 8,
      "column": 32
    },
    "end": {
      "line": 8,
      "column": 64
    }
  },
  "description": null,
  "mutatorName": "ObjectLiteral",
  "replacement": "{}"
}
```

Passing only the file and location of the `BlockStatement` mutant, would also cause the 2nd mutant to be tested.

Perhaps it would make more sense to treat the locations passed to the `mutationTest`  method as exact locations, instead of an area to search for mutants. Combined with the `mutatorName` i think that would be a consistent way of specifying a single mutant?

**Comment by jaspervdveen** [comment]
You're right. We need this and supporting the ability to test specific mutants—rather than all mutants within a code block—would also benefit editor plugins.

One way to implement this could be by introducing an optional `mutants` property to the `MutationTestParams`. This property would contain a list of specific mutants to test.

If the underlying mutation framework supports stable mutant IDs (i.e., IDs that do not change between runs unless the source code changes), we should also allow specifying mutants by their `id`. This would simplify referencing individual mutants. Stryker does not have stable mutant IDs.

We could make the `mutants` list support either:

- `id` (if available and stable), or
- The combination of `filePath`, `location`, and `mutatorName`. You're right that this trio is sufficient to precisely target a single mutant. 

However, to fully support this, we’d also need to update the [mutation-testing-report-schema](https://github.com/stryker-mutator/mutation-testing-elements/blob/master/packages/report-schema/src/mutation-testing-report-schema.json) and the Stryker-frameworks. Currently, `mutatorName` refers to broader mutation categories (e.g., `"ConditionalExpression"`), but identifying specific mutants would require access to the actual operator name (e.g., `"GreaterThanToTrue"`).

**Comment by rouke-broersma** [comment]
Mutant id is already a required field, it is up to the producing/consuming framework to decide to use this information or not. MSP can simply pass it along since it already has it. Frameworks without stable IDs will simply not use it. 

I don't think any changes are required in mte, only in the specific frameworks.
